src/Map.cpp

```diff
@@ -10,70 +10,74 @@ Map::Map(std::string file_name) {
     }
    
     rf.read(reinterpret_cast<char *>(&bounds.left), sizeof (bounds.left));
-
     rf.read(reinterpret_cast<char *>(&bounds.top), sizeof (bounds.top));
-
     rf.read(reinterpret_cast<char *>(&bounds.width), sizeof (bounds.width));
-
     rf.read(reinterpret_cast<char *>(&bounds.height), sizeof (bounds.height));
 
-    tiles = new std::vector<Tile>();
-
-
-    int room_tile_count;
+    int layer_count;
     rf.read(
-        reinterpret_cast<char *>(&room_tile_count), 
-        sizeof(room_tile_count)
+        reinterpret_cast<char *>(&layer_count), 
+        sizeof(layer_count)
     );
 
-
-    for(auto i = 0; i < room_tile_count; i++) {
-        Tile tile;
+    for(auto i = 0; i < layer_count; i++) {
+        int room_tile_count;
         rf.read(
-            reinterpret_cast<char *>(&tile.rotation), 
-            sizeof (tile.rotation)
+            reinterpret_cast<char *>(&room_tile_count), 
+            sizeof(room_tile_count)
         );
 
-        rf.read(
-            reinterpret_cast<char *>(&tile.tile_map_index), 
-            sizeof (tile.tile_map_index)
-        );
+        std::vector<MapTile> tiles;
 
-        rf.read(
-            reinterpret_cast<char *>(&tile.x), 
-            sizeof (tile.x)
-        );
+        for(auto i = 0; i < room_tile_count; i++) {
+            MapTile tile;
+            rf.read(
+                reinterpret_cast<char *>(&tile.rotation), 
+                sizeof (tile.rotation)
+            );
 
-        rf.read(
-            reinterpret_cast<char *>(&tile.y), 
-            sizeof (tile.y)
-        );
+            rf.read(
+                reinterpret_cast<char *>(&tile.tile_map_index), 
+                sizeof (tile.tile_map_index)
+            );
+
+            rf.read(
+                reinterpret_cast<char *>(&tile.x), 
+                sizeof (tile.x)
+            );
 
-        tiles->push_back(tile);
+            rf.read(
+                reinterpret_cast<char *>(&tile.y), 
+                sizeof (tile.y)
+            );
+
+            tiles.push_back(tile);
+        }
+
+        tile_layers.push_back(TileLayer{i, tiles});
+    }
+
+    int entity_count;
+    rf.read(reinterpret_cast<char *>(&entity_count), sizeof(entity_count));
+    for(auto k = 0; k < entity_count; k++) {
+        
+        EntityType type;
+        rf.read(reinterpret_cast<char *>(&type), sizeof (int));
```

Comment:
The values are probably the same, but for clarity should this be `sizeof(EntityType)`?